ideascube/serveradmin/catalog.py

```diff
@@ -549,28 +557,58 @@ def upgrade_packages(self, ids):
 
             self._catalog['installed'][ipkg.id] = (
                 self._catalog['available'][upkg.id])
-            self._persist_cache()
+            self._persist_catalog()
 
         for handler in used_handlers.values():
             handler.commit()
 
     # -- Manage local cache ---------------------------------------------------
-    def _load_cache(self):
-        if os.path.exists(self._cache_catalog):
-            with open(self._cache_catalog, 'r') as f:
-                self._catalog = yaml.safe_load(f.read())
+    def _load_catalog(self):
+        self._catalog = {'available': {}, 'installed': {}}
+
+        try:
+            with open(self._catalog_cache, 'r') as f:
+                catalog = yaml.safe_load(f.read())
+
+        except FileNotFoundError:
+            # That's ok
+            pass
+
+        else:
+            # yaml.safe_load returns None for an empty file
+            if catalog is not None:
+                if 'available' in catalog and 'installed' in catalog:
+                    # The cache on file is in the old format
+                    # https://github.com/ideascube/ideascube/issues/376
+                    self._catalog = catalog
+
+                else:
+                    self._catalog['available'] = catalog
 
-        # yaml.safe_load returns None for an empty file.
-        if not getattr(self, '_catalog', None):
-            self._catalog = {'installed': {}, 'available': {}}
-            self._persist_cache()
+        try:
+            with open(self._installed_storage, 'r') as f:
+                installed = yaml.safe_load(f.read())
+
+        except FileNotFoundError:
+            # That's ok
+            pass
+
+        else:
+            # yaml.safe_load returns None for an empty file
+            if installed is not None:
+                self._catalog['installed'] = installed
+
+        self._persist_catalog()
 
         self._package_caches = [self._local_package_cache]
-        os.makedirs(self._local_package_cache, exist_ok=True)
 
-    def _persist_cache(self):
-        with open(self._cache_catalog, 'w') as f:
-            f.write(yaml.safe_dump(self._catalog, default_flow_style=False))
+    def _persist_catalog(self):
+        def persist(path, data):
```

Comment:
Given that it's used also somwhere else on te module, I would have made the `persist` method available as a global.